bluetooth/src/commonMain/kotlin/Bluetooth.kt

```diff
@@ -153,11 +146,13 @@ class Bluetooth internal constructor(
         scanMode.value = ScanMode.Stopped
     }
 
-    suspend fun isScanning(): StateFlow<Boolean> {
-        return scanningStateRepo.combine(scanMode) { scanState, filter ->
-            filter is ScanMode.Scan && scanState is ScanningState.Initialized.Enabled.Scanning
-        }.stateIn(this)
-    }
+    suspend fun isScanning() = combine(scanningStateRepo, scanMode) { scanState, scanMode ->
+        scanState is ScanningState.Initialized.Enabled.Scanning && scanMode is ScanMode.Scan
+    }.stateIn(this)
+
+    suspend fun isEnabled() = scanningStateRepo.mapLatest {
+        it is ScanningState.Initialized.Enabled
+    }.stateIn(this)
```

Comment:
We should not keep using this pattern since it creates a new StateFlow for every call.

It should be a (`lazy`) `val`.

Reply:
`stateIn` is suspend method